lib/pbench/test/unit/agent/task/test_tools_kill.py

```diff
@@ -79,29 +96,63 @@ def test_load_found(monkeypatch):
         assert ps.load(mydir, "uuid1") is True
 
     @staticmethod
-    def test_noop_killem(monkeypatch):
+    def test_noop_killem(monkeypatch, capsys):
         mock_kf = MagicMock()
         monkeypatch.setattr(kill, "kill_family", mock_kf)
         ps = kill.PidSource("this.pid", "display this")
         ps.killem(ourecho)
         assert not mock_kf.called
+        captured = capsys.readouterr()
+        assert captured.out == ""
+        assert captured.err == ""
 
     @staticmethod
-    def test_killem(monkeypatch):
-        mock_kf = MagicMock()
+    def test_killem_raises_exception(monkeypatch, capsys):
+        mock_kf = MagicMock(side_effect=Exception("exception"))
         monkeypatch.setattr(kill, "kill_family", mock_kf)
         monkeypatch.setattr(
             "pbench.cli.agent.commands.tools.kill.psutil.Process", MyProcess
         )
         ps = kill.PidSource("this.pid", "display this")
         ps.load(AnotherPath("fake_dir1", "123"), "uuid1")
-        ps.load(AnotherPath("fake_dir2", "456"), "uuid2")
-        ps.load(AnotherPath("fake_dir3", "789"), "uuid3")
+        ps.killem(ourecho)
+        captured = capsys.readouterr()
+        assert (
+            captured.out
+            == "Killing display this PIDs ...\n\tKilling 123 (from fake_dir1)\n"
+        )
+        assert captured.err == "\t\terror killing 123: exception\n"
+
+    @staticmethod
+    def test_killem(monkeypatch, capsys):
+        mock_kf = MagicMock()
+        monkeypatch.setattr(kill, "kill_family", mock_kf)
+        monkeypatch.setattr(
+            "pbench.cli.agent.commands.tools.kill.psutil.Process", MyProcess
+        )
+        ps = kill.PidSource("this.pid", "display this")
+        pid_data = {
+            "uuid1": {"dirname": "fake_dir1", "pid": "123"},
+            "uuid2": {"dirname": "fake_dir2", "pid": "456"},
+            "uuid3": {"dirname": "fake_dir3", "pid": "789"},
+        }
+        for uuid, vals in pid_data.items():
+            ps.load(AnotherPath(vals["dirname"], vals["pid"]), uuid)
         ps.killem(ourecho)
         assert len(mock_kf.mock_calls) == 3
-        assert mock_kf.mock_calls[0][1][0].pid == 123
-        assert mock_kf.mock_calls[1][1][0].pid == 456
-        assert mock_kf.mock_calls[2][1][0].pid == 789
+        idx = 0
+        for uuid in pid_data.keys():
+            assert mock_kf.mock_calls[idx][1][0].pid == int(pid_data[uuid]["pid"])
+            idx += 1
+        # Verify a second call to killem() is a no-op.
+        ps.killem(ourecho)
+        assert len(mock_kf.mock_calls) == len(pid_data)
```

Comment:
Line 149 is a nice touch (as long as we're using the table-driven approach)...shouldn't line 142 be the same?